src/components/views/ViewCampaign.jsx

```diff
@@ -62,10 +62,19 @@ class ViewCampaign extends Component {
       }); // TODO: inform user of error
 
     this.loadMoreMilestones(campaignId);
+  }
+
+  componentWillUnmount() {
+    if (this.donationsObserver) this.donationsObserver.unsubscribe();
+  }
 
+  loadDonations() {
     // Lazy load donations
+    const milestoneIds = this.state.milestones.map(m => m._id);
+
+    if (this.donationsObserver) this.donationsObserver.unsubscribe();
     this.donationsObserver = CampaignService.subscribeDonations(
-      campaignId,
+      [this.state.campaign.id, ...milestoneIds],
```

Comment:
This will not work because only last 50 Milestones are loaded by default. I would prefer if you could modify the feathers as described in the issue.